### PR TITLE
Fix the NavigationPlayground example!

### PR DESCRIPTION
**Pull request by cooperka**
## Motivation

The [NavigationPlayground](https://github.com/react-community/react-navigation/tree/master/examples/NavigationPlayground) example app has been broken for months:
- https://github.com/react-community/react-navigation/issues/2168
- https://github.com/react-community/react-navigation/issues/1754
- https://github.com/react-community/react-navigation/issues/1708
- https://github.com/react-community/react-navigation/issues/1629
- https://github.com/react-community/react-navigation/issues/1609
- https://github.com/react-community/react-navigation/issues/1457
- https://github.com/react-community/react-navigation/issues/1325

See the first issue in that list (#2168) for more details about my specific situation and how I came to this solution.

## Changes

- Re-ran `create-react-native-app`, preserving any custom modifications
- Removed the `projectRoots` flag in `app.json` to fix file resolution
- Added a `postinstall` script to the example app to support Yarn installation
- Updated the Contributors Guide to explain the current setup steps, along with some tips for debugging
- Removed the now-obsolete `rn-cli.config.js` and `transformer.js` files

## Test Plan

Follow the new steps on the [Contributors Guide](https://github.com/cribspot/react-navigation/blob/fix-example/docs/guides/Contributors.md#development) to install locally and test it yourself.

Verified from scratch on Boron LTS (node v6.11.1, npm v3.10.10) and node v8.1.4 (npm v4.6.1).

### CODE REVIEW

**Comment by spencercarli** [comment]
Thank you for this! Could you resolve the conflicts that exist and then I'll get this merged

File: examples/NavigationPlayground/app.json [hunk 1]
```diff
@@ -12,15 +12,13 @@
       "icon": "./assets/icons/react-navigation.png",
       "hideExponentText": false
     },
-    "sdkVersion": "16.0.0",
-    "entryPoint": "./examples/NavigationPlayground/node_modules/react-native-scripts/build/bin/crna-entry.js",
+    "sdkVersion": "18.0.0",
```
Comment (spencercarli) [inline examples/NavigationPlayground/app.json]: Bump to sdk 20

File: examples/NavigationPlayground/package.json [hunk 2]
```diff
@@ -5,31 +5,33 @@
...
-      "/node_modules\/.*\/react/"
+      "/node_modules/.*/react-native/",
+      "/node_modules/.*/react/"
     ],
     "transformIgnorePatterns": [
-      "/node_modules\/(?!react-native|react-navigation)/"
+      "/node_modules/(?!react-native|react-navigation)/"
     ]
+  },
+  "dependencies": {
```
Comment (spencercarli) [inline examples/NavigationPlayground/package.json]: Bump all the dependencies for relevant expo 20 versions

**Comment by cooperka** [comment]
Thanks for the review, I'm out of town for a couple days but I will put it on my list for this weekend!

**Comment by cooperka** [comment]
@spencercarli updated. There were some nasty merge conflicts with https://github.com/react-community/react-navigation/commit/ebf1e2e893bf21a58ba6ee72285003ed1cd6bdb8 but the resulting diff is much cleaner than before.

Old diff: https://github.com/react-community/react-navigation/compare/e4cf72d%5E...c6b10b5
New diff: https://github.com/react-community/react-navigation/compare/24a8b2c%5E...52d0083

Tested and verified on iOS and Android after latest changes. Example app is still broken on master, and this PR still fixes it. Ready for merge!

**Comment by spencercarli** [comment]
Awesome - thanks so much for doing this!

**Pull request merged**